### PR TITLE
MML Collision Events Guide

### PR DESCRIPTION
**Pull request by sachamorgese**
Added MML collision events guide
Light content and style refactoring

### CODE REVIEW

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+       },
+       connectionId: number
+    }
+  }
+  ```
+
+
+## Related attributes
+
+- **collision-interval**: Specifies the time interval (in milliseconds) at which collisions are checked.
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:59]: I think this should go above the events with an explanation that this attribute should be present to enable collision events.

File: createJsonFileFromSchema.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const fs = require("fs");
```
Comment (MarcusLongmuir) [inline createJsonFileFromSchema.js:1]: Unintentional commit

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+The example showcases platform that alters its color whenever a user steps on it.
+
+![Basic Example of element collision](/images/guides/collision-events/basic-example.gif)
+
+
+```html
+<!--Basic Example of element collision-->
+<m-cube
+  id="collision-ground"
+  Y="0.2"
```
[the remark this answers is not in the document]
Comment (sachamorgese) [reply src/content/guides/mml-collide-events-guide/body.mdx:76]: Done

File: createJsonFileFromSchema.js (hunk 2 above)
Comment (sachamorgese) [reply createJsonFileFromSchema.js]: Removed

File: tsconfig.tsbuildinfo [hunk 4]
```diff
@@ -0,0 +1 @@
...,372,337,375,365,380,379,333,334,336,335,60,61,12,13,15,14,2,16,17,18,19,20,21,22,23,3,4,27,24,25,26,28,29,30,5,31,32,33,34,6,38,35,36,37,39,7,40,45,46,41,42,43,44,8,50,47,48,49,51,9,52,53,54,57,55,56,58,10,1,11,59,327,557,553,552,556,555,554,347,348,343,345,344,346,342,544,545,546,299,300,301,298,526,297,525,493,547,548,549,494,497,603,616,614,615,617,618,303,302,304,621,619,622,623,624,625,630,631,632,633,629,495,498,496,634,635,638,637,636,499,527,528,529,639,620,305,602,308,310,309,295,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,312,313,314,315,319,316,317,318,323,321,322,645,646,648,661,649,662,664,663,665,667,666,668,650,660,530,324,429,467,647,307,468,469,320,279,311,470,471],"affectedFilesPendingEmit":[296,543,544,545,546,299,300,301,298,526,297,525,493,547,548,549,494,497,603,616,614,615,617,618,303,302,304,621,619,622,623,624,625,630,631,632,633,629,495,498,496,634,635,638,637,636,499,527,528,529,639,620,305,602,308,310,309,295,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,312,313,314,315,319,316,317,318,323,321,322,645,646,648,661,649,662,664,663,665,667,666,668,650,660,530,324,429,467,647,307,468,469,320,279,311,470,471]},"version":"5.0.4"}
```
[the remark this answers is not in the document]
Comment (sachamorgese) [reply tsconfig.tsbuildinfo]: Removed

File: src/content/guides/mml-collide-events-guide/body.mdx (hunk 1 above)
Comment (sachamorgese) [reply src/content/guides/mml-collide-events-guide/body.mdx]: I moved the "related attributes" section above the "related events" one and added an extra paragraph to the introduction specifying collision-interval is necessary to enable collision events. It's also already explained in the description but let me know if I should rephrase it

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+Collision events provide a dynamic way for users to interact with MML elements, enabling a variety
+of interactive experiences. They are triggered based on user collisions in a 3D space. To activate
+the collisions, it's necessary to set the `collision-interval` attribute on the element.
+
+## Related attributes
+
+- **collision-interval**: Specifies the time interval (in milliseconds) at which collisions are
+  checked. Setting this attribute is mandatory to activate the collision events. A smaller number
+  means more frequent checks, but also more events being sent to the client which can impact
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:11]:
```suggestion
  means more frequent checks, but also more events being sent by the client which can impact
```

File: src/content/guides/mml-collide-events-guide/index.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { DocsGuide } from "@/types/docs-guides";
+
+import body from "./body.mdx";
+
+const collisionEvents: DocsGuide = {
+  title: "Getting Started: Collision Events in MML",
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/index.ts:6]:
```suggestion
  title: "Collision Events",
```

File: src/content/guides/mml-collide-events-guide/index.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  description: "Learn how to leverage the power of the collisions.",
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/index.ts:7]:
```suggestion
  description: "Learn how to make MML content that can react to users colliding with it.",
```

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+  performance.
+
+## Related events
+
+These events are fired depending on the user's actions relative to the element:
+
+- [**collisionstart**](/docs/reference/events/MMLCollisionStartEvent): Triggers when a user collides
+  with the element. The callback receives an event object with the user's connectionId and position,
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:19]:
```suggestion
  with the element. The callback receives an event object with the user's `connectionId` and `position`,
```

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+          z: number
+       },
+       connectionId: number
+    }
+  }
+```
+
+- [**collisionmove**](/docs/reference/events/MMLCollisionMoveEvent): Triggers when a user moves
+  while colliding with an element. The callback receives an event object with the user's
+  connectionId and position, inside the property `detail`.
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:37]:
```suggestion
 `connectionId` and `position`, inside the property `detail`.
```

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+        x: number,
+        y: number,
+        z: number
+     },
+     connectionId: number
+  }
+}
+```
+
+- [**collisionend**](/docs/reference/events/MMLCollisionEndEvent): Triggers when a user stops
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:52]: `collisionend` doesn't receive the `position`; only `connectionId`.

File: src/content/guides/mml-collide-events-guide/index.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  body,
+  linkList: [
+    "Related events",
+    "Related attributes",
+    "Basic example of element collision",
+    "Example with multiple users and disconnection events",
+    "Advanced Example with collisionmove events",
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/index.ts:14]:
```suggestion
    "Advanced example with collisionmove events",
```

(to be consistent with others)

File: src/content/guides/mml-collide-events-guide/body.mdx [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+  collisionGround.addEventListener("collisionstart", handleCollisionStart);
+  collisionGround.addEventListener("collisionend", handleCollisionEnd);
+</script>
+```
+
+In this example we have the same platform as before, as well as an `m-label` that shows a message with the number of users connected. The main logic behind this example is the use of a JavaScript. Set object to store unique `connectionId` values. When a user steps on the platform, the `connectionId` in `detail` is added to the set. When a collision ends or the user disconnects, the corresponding connectionId is removed from the set. The user count is simply the size of this set.
+
+---
+
+## Advanced Example with collisionmove events
```
Comment (MarcusLongmuir) [inline src/content/guides/mml-collide-events-guide/body.mdx:178]:
```suggestion
## Advanced example with collisionmove events
```

**Review by MarcusLongmuir — APPROVED** [review]

**Pull request merged**